docs/fault-proof-alpha/immunefi.md

```diff
@@ -0,0 +1,90 @@
+   agent, ensure that an honest challenger's participation always results in a profitable move, and possibly prevent the need for challengers to respond to every invalid claim within the game.
+1. The fault proof system will be integrated into the bridge contracts, specifically the `OptimismPortal`, in order to enable the system to be used in production and verify the correctness of output roots that withdrawals are proven against.
+
+## Bounty Scope
+The scope of the bounty is limited to the fault proof alpha system. This includes the following components, in order of security review priority:
+1. **Cannon**: The [Cannon][cannon] binary and its dependencies.
+1. **op-program**: The [`op-program`][op-program] binary and its dependencies.
+1. **Smart Contracts**
+    1. The [Cannon][cannon-contracts] contracts and their dependencies.
+    1. The [Dispute Game][dispute-game] and their dependencies.
+1. **op-challenger**: The [`op-challenger`][op-challenger] binary and its dependencies.
+
+As mentioned above in the "[Plans for the next iteration](#plans-for-the-next-iteration)" section, there will soon be a number of large architectural changes to the [dispute smart contracts][dispute-game]
+as well as the [`op-challenger`][op-challenger] in order to support the features that will bring the system to a production ready state. During this time, it is unlikely that [Cannon][cannon], the [Cannon contracts][cannon-contracts],
+or the [`op-program`][op-program] will change significantly, and as such, we recommend focusing efforts primarily on these components.
+
+There are several key invariants that must be maintained in order for the system to be considered secure. A bounty report must demonstrate a bug which breaks one of these invariants.
+1. **Cannon**
+    1. [Cannon][cannon]'s `mipsevm` must be functionally equivalent to the [MIPS thread context][cannon-contracts] implemented in Solidity. Any disparities that result in different `op-program` execution are a bug.
+        1. Both [Cannon][cannon] and the on-chain [MIPS thread context][cannon-contracts] must produce the same output given an identical setup state and input data.
+    1. Both [Cannon][cannon] and the on-chain [MIPS thread context][cannon-contracts] must produce a deterministic output given an identical setup state and input data.
+    1. Both [Cannon][cannon] and the on-chain [MIPS thread context][cannon-contracts] must never panic on a state transition with honest input data / setup state.
+        1. Note: There are a number of instructions from MIPS that Cannon does not support. Specifically, this invariant covers panic conditions within the realm of supported instructions and valid honest input data / setup state where cannon otherwise should have completed execution and produced a valid/invalid opinion about the state transition.
```

Comment:
```suggestion
        1. Note: There are a number of instructions from MIPS, and system calls in Linux, that Cannon does not support. Specifically, this invariant covers panic conditions within the realm of supported instructions and valid honest input data / setup state where cannon otherwise should have completed execution and produced a valid/invalid opinion about the state transition. The op-program may contain "dead code", non-reachable invalid instructions that do not affect the output.
```